test/integration/puppetlabs/services/jruby/class_info_test.clj

```diff
@@ -0,0 +1,229 @@
+(ns puppetlabs.services.jruby.class-info-test
+  (:require [clojure.test :refer :all]
+            [puppetlabs.kitchensink.core :as ks]
+            [puppetlabs.services.jruby.jruby-testutils :as jruby-testutils]
+            [puppetlabs.services.jruby.jruby-puppet-internal :as jruby-internal]
+            [me.raynes.fs :as fs]
+            [cheshire.core :as cheshire])
+  (:import (java.util.concurrent LinkedBlockingDeque)))
+
+(deftest ^:integration class-info-test
+  (testing "class info properly enumerated for"
+    (let [pool (LinkedBlockingDeque. 1)
```

Comment:
This `let` block is obviously pretty massive.  Even though they don't get used outside of this `deftest` block, I could move some of the "helper" type functions out to separate `defn`s just to shorten this up.

The `testing` sub-blocks are intentionally ordered the way they are to validate that the nature of what class information is extracted from parsing is updated appropriately as the underlying content on disk changes from one call to the next.  By design, then, the `testing` sub-blocks aren't intended to be run independently.